common/reflect_resource.go

```diff
@@ -629,8 +639,16 @@ func collectionToMaps(v any, s *schema.Schema, aliases map[string]map[string]str
 					return fmt.Errorf("%s: %v", path, err)
 				}
 				data[fieldName] = nv
+			case schema.TypeBool, schema.TypeInt:
+				// If the field is not in the force send fields, we should not store it in the state
+				// Only bool and int values are in force send fields now
+				if len(forceSendFields) > 0 && !slices.Contains(forceSendFields, valueField.sf.Name) {
```

Comment:
Is the `len` check needed?

Reply:
@mgyucht yes, if there are no forceSendFields set we don't want to execute this check because otherwise no fields  will be written to TF state at all